pkg/svcgen/params/builder.go

```diff
@@ -0,0 +1,275 @@
+			}
+			basicMessages[0] = resource
+			basicMessages[3] = &MethodMessage{
+				Name:   reqName,
+				Fields: []MethodMessageField{{Name: id, Type: "string", Tag: 1}},
+			}
+		case "create":
+			methodName := "Create" + name.singularCamel
+			reqName := methodName + "Request"
+			basicMethods[2] = &MethodParams{
+				Method:         methodName,
+				requestCommon:  reqName,
+				responseCommon: resource.Name,
+				HTTP:           MethodHTTPParams{Method: "post", Path: name.pluralSnake, Body: name.singularSnake},
+			}
+			basicMessages[0] = resource
+			basicMessages[4] = &MethodMessage{
+				Name:   reqName,
+				Fields: []MethodMessageField{{Name: name.singularSnake, Type: name.singularCamel, Tag: 1}},
+			}
+		case "update":
+			methodName := "Update" + name.singularCamel
+			reqName := methodName + "Request"
+			basicMethods[3] = &MethodParams{
+				Method:         methodName,
+				requestCommon:  reqName,
+				responseCommon: resource.Name,
+				HTTP:           MethodHTTPParams{Method: "patch", Path: name.pluralSnake + "/{" + name.singularSnake + "." + id + "}", Body: name.singularSnake},
+			}
+			basicMessages[0] = resource
+			basicMessages[5] = &MethodMessage{
+				Name:   reqName,
+				Fields: []MethodMessageField{{Name: name.singularSnake, Type: name.singularCamel, Tag: 1}},
+			}
+		case "delete":
+			methodName := "Delete" + name.singularCamel
+			reqName := methodName + "Request"
+			basicMethods[4] = &MethodParams{
+				Method:        methodName,
+				requestCommon: reqName,
+				responseProto: "google.protobuf.Empty",
+				responseGo:    "empty.Empty",
+				HTTP:          MethodHTTPParams{Method: "delete", Path: name.pluralSnake + "/{" + id + "}"},
+			}
+			basicMessages[6] = &MethodMessage{
+				Name:   reqName,
+				Fields: []MethodMessageField{{Name: id, Type: "string", Tag: 1}},
+			}
+			params.ProtoImports = append(params.ProtoImports, "google/protobuf/empty.proto")
+			params.GoImports = append(params.GoImports, "github.com/golang/protobuf/ptypes/empty")
+		default:
+			methodName := snaker.SnakeToCamel(meth)
+			reqName := methodName + "Request"
+			respName := methodName + "Response"
+			customMethods = append(customMethods, MethodParams{
+				Method:         methodName,
+				requestCommon:  reqName,
+				responseCommon: respName,
+				HTTP:           MethodHTTPParams{Method: "get", Path: name.pluralSnake + "/" + snaker.CamelToSnake(meth)},
+			})
+			customMessages = append(
+				customMessages,
+				MethodMessage{Name: reqName},
+				MethodMessage{Name: respName},
+			)
+		}
+	}
+
+	for _, meth := range basicMethods {
+		if meth != nil {
+			params.Methods = append(params.Methods, *meth)
+		}
+	}
+	for _, msg := range basicMessages {
+		if msg != nil {
+			params.Messages = append(params.Messages, *msg)
+		}
+	}
+	for _, meth := range customMethods {
+		params.Methods = append(params.Methods, meth)
+	}
+	for _, msg := range customMessages {
```

Comment:
**[megacheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
 should replace loop with params.Messages = append(params.Messages, customMessages...) (S1011)